test/e2e/etcdctl_test.go

```diff
@@ -0,0 +1,116 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/openshift/cluster-etcd-operator/test/e2e/framework"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TestEtcdctlCommands executes all known etcdctl commands inside of the etcdctl container.
+// The test is not intended to be a functional test yet a sanity test that the container
+// ENV is populated correctly and that etcdctl consumes that ENV properly.
+func TestEtcdctlCommands(t *testing.T) {
+	// setup
+	cs := framework.NewClientSet("")
+	pod, err := cs.CoreV1Interface.Pods("").List(context.TODO(), metav1.ListOptions{LabelSelector: "k8s-app=etcd"})
+	require.NoError(t, err)
+	podName := pod.Items[0].Name
+
+	testCases := []struct {
+		command       string
+		skip          bool
+		skipReason    string
+		expectedError string
+	}{
+		{"etcdctl alarm disarm", false, "", ""},
+		{"etcdctl alarm list", false, "", ""},
+		{"etcdctl del --prefix /etcdctl-check-perf/", false, "", ""},
+		{"etcdctl check perf", false, "", ""},
+		{"etcdctl compaction 0", false, "", "Error: etcdserver: mvcc: required revision has been compacted"},
+		{"etcdctl defrag", false, "", ""},
+		{"etcdctl elect myelection", false, "", "no proposal argument but -l not set"},
+		// endpoint
+		{"etcdctl endpoint hashkv", false, "", ""},
+		{"etcdctl endpoint health", false, "", ""},
+		{"etcdctl endpoint status", false, "", ""},
+		{"etcdctl get /foo", false, "", ""},
+		// lease
+		{"etcdctl lease grant 10", false, "", ""},
+		{"etcdctl lease keep-alive 694d74da6ca8020b", false, "", ""},
+		{"etcdctl lease list", false, "", ""},
+		{"etcdctl lease revoke 694d74da6ca8020b", false, "", "etcdserver: requested lease not found"},
+		{"etcdctl lease timetolive 694d74da6ca8020b", false, "", ""},
+		// member
+		{"etcdctl member list", false, "", ""},
+		{"etcdctl member add tester", false, "", "member peer urls not provided"},
+		{"etcdctl member promote 8e9e05c52164694d", false, "", "etcdserver: member not found"},
+		{"etcdctl member remove 8e9e05c52164694d", false, "", "etcdserver: member not found"},
+		{"etcdctl member update 8e9e05c52164694d", false, "", "member peer urls not provided"},
+		{"etcdctl move-leader 8e9e05c52164694d", false, "", ""},
+		{"etcdctl put /etcdctl-check-perf/foo bar", false, "", ""},
+		{"etcdctl snapshot save ./backup.db", false, "", ""},
+		{"etcdctl snapshot status ./backup.db", false, "", ""},
+		{"etcdctl snapshot restore ./backup.db", false, "", ""},
```

Comment:
Are these potentially disruptive?

Reply:
snapshot save could have overhead but it should not be disruptive. the rest would not have disruptive cost.